totem/blog/mobile_api.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from typing import List, Optional
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from ninja import Field, FilterSchema, ModelSchema, Router, Schema
+
+from totem.users.models import User
+from .models import BlogPost
+
+from ninja import ModelSchema
+
+class AuthorSchema(ModelSchema):
+    class Meta:
+        model = User
+        fields = ["name", "slug"]
+
+
+class BlogPostSchema(ModelSchema):
+    author: Optional[AuthorSchema] = None
+    header_image_url: Optional[str] = None
+
+    @staticmethod
+    def resolve_header_image_url(obj) -> Optional[str]:
+        if obj.header_image and hasattr(obj.header_image, 'url'):
+            return obj.header_image.url
+        return None
+
+    class Meta:
+        model = BlogPost
+        fields = ["title", "subtitle", "content", "date_published", "slug", "author", "publish"]
```

Comment:
Might be able to use "content_html" instead of "content" here to get the rendered HTML instead of raw Markdown. The property is in the "MarkdownMixin".

Reply:
For some reason, django doesn't recognize this property. I added a property in the model and a static "resolve_" method.